src/components/AnotherComponent.jsx

```diff
@@ -0,0 +1,55 @@
+import React, { useEffect, useState, useContext } from 'react';
+import axios from 'axios';
+import { LinkContext } from '../context/LinkContext';
+
+const AnotherComponent = () => {
+  const { selectedLink } = useContext(LinkContext);
+  const [data, setData] = useState(null);
+
+  useEffect(() => {
+    if (selectedLink) {
+      fetchData(selectedLink);
+    }
+  }, [selectedLink]);
+
+  const fetchData = async (url) => {
+    try {
+      const response = await axios.get(url);
+      console.log(response, "data");
```

Comment:
always remove console.log statements used for debugging. try to avoid using console.logs, prefer vscode/chrome debugger.